### PR TITLE
[docs] refactor content to focus on React usage

### LINKED ISSUE #2465: [Docs] Move JS sections first, before CSS

### LINKED ISSUE #2542: Documentation problem; DateRangeInput does not show required props
the line example for [DateRangeInput](http://blueprintjs.com/docs/v2/#datetime/daterangeinput) does not have the two required props, `formatDate` and `parseDate`. Trying to use that code will result in errors.
I'd be very happy to just fix this myself, but I'm not actually sure where in the repo the documentation lives to submit a pr.

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- refactor **Core Components & Overlays** documentation content to focus on React usage (**Forms** and other packages will happen in separate PRs)

### CODE REVIEW

File: packages/core/src/components/button/button-group.md [hunk 1]
```diff
@@ -2,7 +2,44 @@
...
 
-@## CSS API
+@reactExample ButtonGroupExample
+
+@## Usage with popovers
+
+`Button`s inside a `ButtonGroup` can optionally be wrapped with a [`Popover`](#core/components/popover). The following example demonstrates this composition:
+
+```tsx
+<ButtonGroup className={Classes.ALIGN_LEFT}>
```
Comment (giladgray) [inline packages/core/src/components/button/button-group.md]: there's an `alignText` prop

Comment (giladgray) [reply packages/core/src/components/button/button-group.md]: tho i'm not convinced this code sample is helpful at all.

File: packages/core/src/components/button/button-group.md [hunk 2]
```diff
@@ -2,7 +2,44 @@
...
+
+@reactExample ButtonGroupPopoverExample
+
+@## Props
+
+This component is a simple wrapper around the CSS API.
+It exposes shorthand props for CSS modifier classes and supports the full range of HTML props.
+
+```tsx
+<ButtonGroup minimal={true} large={false} onMouseEnter={...}>
```
Comment (giladgray) [inline packages/core/src/components/button/button-group.md]: `large={false}` is default, remove

File: packages/core/src/components/button/button-group.md [hunk 3]
```diff
@@ -34,48 +71,8 @@ left-align button text and icon and right-align `rightIcon`.
...
-
-@reactExample ButtonGroupPopoverExample
-
 <div class="@ns-callout @ns-intent-primary @ns-icon-info-sign">
-    In vertical button groups, button content will be centered by default. You can align button content to the left or right using `@ns-align-left` and `@ns-align-right`, respectively.
+    In vertical button groups, button content will be centered by default. You can align button content to the left or right using `.@ns-align-left` and `.@ns-align-right`, respectively.
 </div>
 
-@interface IButtonGroupProps
+@css button-group-vertical
```
Comment (giladgray) [inline packages/core/src/components/button/button-group.md]: these css examples should be refactored

File: packages/core/src/components/card/card.md [hunk 4]
```diff
@@ -21,24 +39,3 @@ changes to a pointer and increases the elevation shadow on the card.
 Users expect an interactive card to be a single clickable unit.
 
 @css card-interactive
```
Comment (giladgray) [inline packages/core/src/components/card/card.md]: can merge this with other css example

File: packages/core/src/components/collapsible-list/collapsible-list.md [hunk 5]
```diff
@@ -7,7 +7,7 @@ props from the `MenuItem` children.
 
 @reactExample CollapsibleListExample
 
-@## JavaScript API
+@## Props
 
 The `CollapsibleList` component is available in the __@blueprintjs/core__ package.
```
Comment (giladgray) [inline packages/core/src/components/collapsible-list/collapsible-list.md]: oops

File: packages/core/src/components/collapsible-list/collapsible-list.md [hunk 6]
```diff
@@ -18,7 +18,9 @@ items using their [`IMenuItemProps`](#core/components/menu.menu-item).
 
 @interface ICollapsibleListProps
 
-@## Separators
+@## CSS
+
+### Separators
```
Comment (giladgray) [inline packages/core/src/components/collapsible-list/collapsible-list.md]: `@### Separators` - it's not a full-fledged CSS API section (cuz the component is React only) but rather a usage note

File: packages/core/src/components/editable-text/editable-text.md [hunk 7]
```diff
@@ -62,3 +56,5 @@ _vertically_ instead, based on the number of lines of text. You can use the `min
     which would provide a sub-optimal experience for users (multiline text does not always render
     cleanly into a single line).
 </div>
+
+@interface IEditableTextProps
```
Comment (giladgray) [inline packages/core/src/components/editable-text/editable-text.md]: this should be in the `Props` section directly.

File: packages/core/src/components/tag/tag.md [hunk 8]
```diff
@@ -43,3 +23,19 @@ Blueprint class name.
...
+A simple `.@ns-tag` without the remove button can easily function as a badge.
+
+@css tag
+
+@### Minimal tags
+
+Add the `.@ns-minimal` modifier for a lighter tag appearance. The translucent background color
+will adapt to its container's background color.
+
+@css tag-minimal
```
Comment (giladgray) [inline packages/core/src/components/tag/tag.md]: merge with other CSS example

File: packages/core/src/components/text/text.md [hunk 9]
```diff
@@ -5,7 +5,7 @@ conditionally adding the title attribute and truncating with an ellipsis when co
 
 @reactExample TextExample
 
-@## JavaScript API
+@## Props
 
 The `Text` component is available in the __@blueprintjs/core__ package.
```
Comment (giladgray) [inline packages/core/src/components/text/text.md]: remove

File: packages/core/src/components/editable-text/editable-text.md [hunk 10]
```diff
@@ -20,41 +20,41 @@ You should not use `EditableText` when a static always-editable `<input>` or
...
-
 Users confirm text in multiline mode by pressing `ctrl` `enter` or `cmd` `enter` rather than
 simply `enter`. (Pressing the `enter` key by itself moves the cursor to the next line.)
 
 Additionally, in multiline mode the component's width is fixed at 100%. It grows or shrinks
 _vertically_ instead, based on the number of lines of text. You can use the `minLines` and
 `maxLines` props to constrain the vertical size of the component.
 
-<div class="@ns-callout @ns-intent-warning @ns-icon-warning-sign">
+<div class="@ns-callout @ns-intent-warning @ns-icon-wa`rning-sign">
```
Comment (giladgray) [inline packages/core/src/components/editable-text/editable-text.md]: oops backtick

File: packages/core/src/components/popover/popover.md [hunk 11]
```diff
@@ -299,7 +296,7 @@ everything else on the page without needing to manually adjust z-indices, and Po
 
 @reactExample PopoverPortalExample
 
-@## Style
+@## CSS
```
Comment (giladgray) [inline packages/core/src/components/popover/popover.md]: ❌ revert to Style

File: packages/core/src/components/portal/portal.md [hunk 12]
```diff
@@ -5,10 +5,7 @@ hierarchy. It is an essential piece of [`Overlay`](#core/components/overlay), re
 the overlay contents cover the application below. In most cases you do not need to use a `Portal`
 directly; this documentation is provided simply for reference.
 
-@## JavaScript API
-
-The `Portal` component is available in the __@blueprintjs/core__ package. Make sure to review the
-[getting started docs for installation info](#blueprint/getting-started).
+@## Props
```
Comment (giladgray) [inline packages/core/src/components/portal/portal.md]: props should come after other usage sections

File: packages/core/src/components/tabs/tabs.md [hunk 13]
```diff
@@ -1,21 +1,8 @@
...
-JavaScript component does this by default).
-
-@css tabs
-
-@## JavaScript API
+@reactExample TabsExample
 
-The `Tabs` and `Tab` components are available in the __@blueprintjs/core__
-package. Make sure to review the [getting started docs for installation info](#blueprint/getting-started).
+@## Usage
```
Comment (giladgray) [inline packages/core/src/components/tabs/tabs.md]:
```
Tabs
   Props
      Tab
   CSS
```
?

File: packages/core/src/components/tag/tag.md [hunk 14]
```diff
@@ -1,36 +1,20 @@
...
 
-Tag components render `.@ns-tag` elements with optional close buttons. Provide tag content as `children`.
+@## CSS
 
-You can provide your own props to these components as if they were regular JSX HTML elements. If
-you provide a `className` prop, the class names you provide will be added alongside of the default
-Blueprint class name.
-
-@interface ITagProps
+Create a tag with a `span.@ns-tag`. An optional "remove" button can be added
```
Comment (benjdj6) [inline packages/core/src/components/tag/tag.md]: Bold remove as it is an interactive UI element.

Comment (giladgray) [reply packages/core/src/components/tag/tag.md]: the element doesn't actually say "remove", it's just an ✖️ . does it still need bold?

File: packages/core/src/components/portal/portal.md [hunk 15]
```diff
@@ -1,31 +1,21 @@
...
 
-<div class="@ns-callout @ns-intent-warning @ns-icon-warning-sign">
-    <h4 class="@ns-heading">A note about responsive layouts</h4>
-    For a single-page app, if the `<body>` is styled with `width: 100%` and `height: 100%`, a `Portal`
-    may take up extra whitespace and cause the window to undesirably scroll. To fix this, instead
-    apply `position: absolute` to the `<body>` tag.
-</div>
+1. `Portal` `children` are wrapped in an extra `<div>` inside the portal container element.
+1. Test harnesses such as `enzyme` cannot trivially find elements "through" Portals as they're not in the same React tree.
+1. React `context` _is_ preserved (this one's a good thing).
```
[the remark this answers is not in the document]
Comment (giladgray) [reply packages/core/src/components/portal/portal.md]: markdown lists tho. keeping them all the same number is more maintainable for future edits

File: packages/core/src/components/tag/tag.md (hunk 14 above)
Comment (benjdj6) [reply packages/core/src/components/tag/tag.md]: It does not.

File: packages/core/src/components/portal/portal.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (benjdj6) [reply packages/core/src/components/portal/portal.md]: Forgot that markdown was forgiving about this (used to rst).

**Pull request merged**